File: perl/substr_count.pl
use strict;
use warnings;
use Data::Dumper;
use Test::More;

sub title
{
  my $v = shift;
  print "\n\n\n\n$v\n\n";
}

sub var_dump
{
  my $v = shift;
  print Dumper $v;
}

sub observation
{
  my $v = shift;
  print "\n$v";
}



my $s0 = "";
my $s1 = "a";
my $s2 = "a\n";
my $s3 = "a\nb";
my $s4 = "\nb";



print "Content-Type: text/html\n\n";
print "<pre>";

title("count empty string");
my $res0 = () = $s0 =~ /\n/g;
var_dump($res0);

title("count no occurrences");
my $res1 = () = $s1 =~ /\n/g;
var_dump($res1);

title("count as sufix");
my $res2 = () = $s2 =~ /\n/g;
var_dump($res2);

title("count as infix");
my $res3 = () = $s3 =~ /\n/g;
var_dump($res3);

title("count as prefix");
my $res4 = () = $s4 =~ /\n/g;
var_dump($res4);

print "</pre>";

File: js/substr_count.js
function title(v)
{
  console.log("-");
  console.log(v);
}

function observation(v)
{
  console.log(v);
}

function var_export(v)
{
  console.log(v);
}

var s0 = "";
var s1 = "a";
var s2 = "a\n";
var s3 = "a\nb";
var s4 = "\nb";

title('count empty string');
var res0 = s0.split("\n").length - 1;
var_export(res0);

title('count no occurrences');
var res1 = s1.split("\n").length - 1;
var_export(res1);

title('count as suffix');
var res2 = s2.split("\n").length - 1;
var_export(res2);

title('count as infix');
var res3 = s3.split("\n").length - 1;
var_export(res3);

title('count as prefix');
var res4 = s4.split("\n").length - 1;
var_export(res4);

